pkg/pillar/cmd/nim/serverdns.go

```diff
@@ -0,0 +1,190 @@
+package nim
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"time"
+
+	dns "github.com/Focinfi/go-dns-resolver"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+const (
+	minTTLSec       int = 30
+	maxTTLSec       int = 3600
+	extraSec        int = 10
+	etcHostFileName     = "/etc/hosts"
+	tmpHostFileName     = "/tmp/etchosts"
+	resolvFileName      = "/etc/resolv.conf"
+)
+
+// go routine for dns query to the controller
+func queryCloudDNS(ps *pubsub.PubSub) {
```

Comment:
I think having "cloud" in many of these new names is confusing. As the comment says on L25 it is about the controller.
So wouldn't it make sense to replace "cloud" in the function and variable names with "controller"?

Also, this file is called "serverdns.go"; should that be "controllerdns.go" to be consistent? (I know it reads a file called /config/server, but can't change that without lots of compatibility concerns.)